RecoEgamma/EgammaTools/src/HGCalIsoCalculator.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * HGCalIsoCalculator.cc
+ *
+ *  Created on: 13 Oct 2017
+ *      Author: jkiesele, ncsmith
+ */
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "RecoEgamma/EgammaTools//interface/HGCalIsoCalculator.h"
+
+HGCalIsoCalculator::HGCalIsoCalculator():dr2_(0.15*0.15),mindr2_(0),rechittools_(nullptr),debug_(false),nlayers_(30){
+    setNRings(5);
+}
+
+HGCalIsoCalculator::~HGCalIsoCalculator(){
+}
+
+void HGCalIsoCalculator::setRecHits(
+        edm::Handle<HGCRecHitCollection> hitsEE,
+        edm::Handle<HGCRecHitCollection> hitsFH,
+        edm::Handle<HGCRecHitCollection> hitsBH
+     ){
+    recHitsEE_=hitsEE;
+    recHitsFH_=hitsFH;
+    recHitsBH_=hitsBH;
+
+    if(!rechittools_)
+        throw cms::Exception("HGCalIsoCalculator::produceHGCalIso: rechittools not set");
+    
+    hitEtaPhiCache_.clear();
```

Comment:
perhaps its good to also provide a clear function so that memory can be cleared when its no longer needed by the calling code

Reply:
This is called by `HGCalEgammaIDHelper::eventInit()` and lasts the duration of an event.  One could provide a clear in the calling class too, but anyway memory will stay allocated since the class member is not destructed and `clear()` is not guaranteed to deallocate.  Forcing the `free` seems pointless since the next event will probably have a similar number of RecHits.

Reply:
p.s. it's about 4 Mb / thread